test/unit/resources/aws_vpc_subnet_test.rb

```diff
@@ -0,0 +1,148 @@
+    assert_nil(AwsVpcSubnet.new(vpc_id: 'vpc-00000000', subnet_id: 'subnet-00000000').availability_zone)
+  end
+
+  def test_property_available_ip_address_count
+    assert_equal(251, AwsVpcSubnet.new(vpc_id: 'vpc-12345678', subnet_id: 'subnet-12345678').available_ip_address_count)
+    assert_nil(AwsVpcSubnet.new(vpc_id: 'vpc-00000000', subnet_id: 'subnet-00000000').available_ip_address_count)
+  end
+
+  def test_property_default_for_az
+    assert_equal(false, AwsVpcSubnet.new(vpc_id: 'vpc-12345678', subnet_id: 'subnet-12345678').default_for_az)
+    assert_nil(AwsVpcSubnet.new(vpc_id: 'vpc-00000000', subnet_id: 'subnet-00000000').default_for_az)
+  end
+
+  def test_property_map_public_ip_on_launch
+    assert_equal(false, AwsVpcSubnet.new(vpc_id: 'vpc-12345678', subnet_id: 'subnet-12345678').map_public_ip_on_launch)
+    assert_nil(AwsVpcSubnet.new(vpc_id: 'vpc-00000000', subnet_id: 'subnet-00000000').map_public_ip_on_launch)
+  end
+
+  def test_property_state
+    assert_equal('available', AwsVpcSubnet.new(vpc_id: 'vpc-12345678', subnet_id: 'subnet-12345678').state)
+    assert_nil(AwsVpcSubnet.new(vpc_id: 'vpc-00000000', subnet_id: 'subnet-00000000').state)
+  end
+
+  def test_property_ipv_6_cidr_block_association_set
+    assert_equal([], AwsVpcSubnet.new(vpc_id: 'vpc-12345678', subnet_id: 'subnet-12345678').ipv_6_cidr_block_association_set)
+    assert_nil(AwsVpcSubnet.new(vpc_id: 'vpc-00000000', subnet_id: 'subnet-00000000').ipv_6_cidr_block_association_set)
+  end
+
+  def test_property_assign_ipv_6_address_on_creation
+    assert_equal(false, AwsVpcSubnet.new(vpc_id: 'vpc-12345678', subnet_id: 'subnet-12345678').assign_ipv_6_address_on_creation)
+    assert_nil(AwsVpcSubnet.new(vpc_id: 'vpc-00000000', subnet_id: 'subnet-00000000').assign_ipv_6_address_on_creation)
+  end
+end
+
+
+#=============================================================================#
+#                               Test Fixtures
+#=============================================================================#
+
+module AwsMVSSB
+  class Basic < AwsVpcSubnet::Backend
+    def describe_subnets(query)
+      subnets = {
+        'vpc-12345678' => OpenStruct.new({
+          :subnets => [
+            OpenStruct.new({
+              availability_zone: "us-east-1",
+              available_ip_address_count: 251,
+              cidr_block: "10.0.1.0/24",
+              default_for_az: false,
+              map_public_ip_on_launch: false,
+              state: "available",
+              subnet_id: "subnet-12345678",
+              vpc_id: "vpc-12345678",
+              ipv_6_cidr_block_association_set: [],
+              assign_ipv_6_address_on_creation: false,
+            }),
+          ]
+        }),
+        'empty' => OpenStruct.new({
+          :subnets => []
+        })
+      }
+
+      return subnets[query[:filters][0][:values][0]] unless subnets[query[:filters][0][:values][0]].nil?
+      subnets['empty']
```

Comment:
Hmm, you're catching exceptions in one place, which means you think it may throw an exception on a miss.  If so, the test fixture here should throw an exception, too.